clang/lib/CIR/CodeGen/CIRGenBuiltinNVPTX.cpp

```diff
@@ -0,0 +1,82 @@
+//===---- CIRGenBuiltinX86.cpp - Emit CIR for X86 builtins ----------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This contains code to emit NVPTX Builtin calls.
+//
+//===----------------------------------------------------------------------===//
+
+#include "CIRGenCXXABI.h"
+#include "CIRGenCall.h"
+#include "CIRGenFunction.h"
+#include "CIRGenModule.h"
+#include "TargetInfo.h"
+#include "clang/CIR/MissingFeatures.h"
+
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/IR/Value.h"
+#include "clang/AST/GlobalDecl.h"
+#include "clang/Basic/Builtins.h"
+#include "clang/Basic/TargetBuiltins.h"
+#include "clang/CIR/Dialect/IR/CIRDialect.h"
+#include "clang/CIR/Dialect/IR/CIRTypes.h"
+#include "llvm/Support/ErrorHandling.h"
+
+using namespace clang;
+using namespace clang::CIRGen;
+using namespace cir;
+
+mlir::Value CIRGenFunction::emitNVPTXBuiltinExpr(unsigned BuiltinID,
+                                                 const CallExpr *E) {
+  auto GetIntrinsic = [&](const char *name) {
```

Comment:
This lambda should be lower case: https://llvm.github.io/clangir/GettingStarted/coding-guideline.html